### PR TITLE
docs(istio): istio guide - k8s 1.18, istio 1.6.7

### LINKED ISSUE #703: docs: how to integrate with Istio
Kong and Istio can play together. We have written blogs about these, but these should be included in documentation:
https://konghq.com/blog/kong-istio-setting-service-mesh-kubernetes-kiali-observability/
https://kubernetes.io/blog/2020/03/18/kong-ingress-controller-and-istio-service-mesh/

### PR DESCRIPTION
**Pull request by mflendrich**
**What this PR does / why we need it**:

Defines a Getting Started guide using KIC and Istio 1.6.7, deploying a sample application, and seeing the data collected by Istio.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

closes #703 #777

**Special notes for your reviewer**:

Please consider whether linking it immediately under _Getting started_ is appropriate

### CODE REVIEW

**Comment by mflendrich** [comment]
[[mflendrich] Docs on setting up Kong Ingress Controller with Istio](https://trello.com/c/6NVa5Deu/357-mflendrich-docs-on-setting-up-kong-ingress-controller-with-istio)

File: docs/guides/getting-started-istio.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
+## Running Kong Ingress Controller with Istio
+
+By following this guide, you will:
+* install Istio v1.6.7 and Kong in your cluster,
+* deploy an example Istio-enabled application (_bookinfo_),
+* deploy an `Ingress` customized with a `KongPlugin` for the example application,
+* make several requests to the sample application via Kong and Istio,
+* see the performance metrics of the sample application, provided by Istio.
```
Comment (rainest) [inline docs/guides/getting-started-istio.md:8]:
```suggestion
In this guide, you will:
* Install Istio v1.6.7 and Kong in your cluster.
* Deploy an example Istio-enabled application (_bookinfo_).
* Deploy an `Ingress` customized with a `KongPlugin` for the example application.
* Make several requests to the sample application via Kong and Istio.
* See the performance metrics of the sample application, provided by Istio.
```
See https://docs.microsoft.com/en-us/style-guide/scannable-content/lists

Why the convention exists I don't really know, but normally the bullets are treated as quasi-independent sentences and are styled as such.

File: docs/guides/getting-started-istio.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+### Prerequisites
+
+* A Kubernetes v1.15-1.18 cluster which can pull container images from public registries. Examples:
+    * A managed Kubernetes cluster (AWS EKS, Google Cloud GKE, Azure AKS),
+    * Minikube,
+    * `microk8s` with the `dns` addon enabled,
+* `kubectl` with admin access to the cluster.
```
Comment (rainest) [inline docs/guides/getting-started-istio.md:16]:
```suggestion
For this guide, you will need:

* A Kubernetes v1.15 (or newer) cluster which can pull container images from public registries. For example, you can use:
    * A managed Kubernetes cluster (AWS EKS, Google Cloud GKE, Azure AKS).
    * Minikube.
    * `microk8s` with the `dns` addon enabled.
* `kubectl` with admin access to the cluster.
```
Unbounded the version requirements, as I don't expect we'll see issues where newer K8S versions are no longer compatible. Cross that bridge when we get there if so.

File: docs/guides/getting-started-istio.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+### Download Istio
+
+Download the Istio bundle at version 1.6.7.
```
Comment (rainest) [inline docs/guides/getting-started-istio.md:20]:
```suggestion
Download the Istio bundle at version 1.6.7:
```

File: docs/guides/getting-started-istio.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+Istio 1.6.7 Download Complete!                                                                                                 
+                                                               
+Istio has been successfully downloaded into the istio-1.6.7 folder on your system.                                                                                                                                                                            
+...
+...
+```
+
+### Install Istio Operator
+
+Invoke `istioctl` to deploy the Istio Operator to the Kubernetes cluster.
```
Comment (rainest) [inline docs/guides/getting-started-istio.md:35]:
```suggestion
Invoke `istioctl` to deploy the Istio Operator to the Kubernetes cluster:
```

File: docs/guides/getting-started-istio.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+```console
+$ ./istio-1.6.7/bin/istioctl operator init
+Using operator Deployment image: docker.io/istio/operator:1.6.7
+✔ Istio operator installed                                                                                                                                                                                                                                    
+✔ Installation complete
+```
+
+### Deploy Istio using Operator
+
+Deploy Istio using Istio Operator.
```
Comment (rainest) [inline docs/guides/getting-started-istio.md:46]:
```suggestion
Deploy Istio using Istio Operator:
```

File: docs/guides/getting-started-istio.md [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+NAMESPACE: kong-istio
+STATUS: deployed
+...
+```
+
+_Optional:_ Run `kubectl describe pod -n kong-istio -l app.kubernetes.io/instance=example-kong` to see that the Istio sidecar (`istio-proxy`) is running alongside Kong Ingress Controller.
+
+### Deploy bookinfo in an Istio-enabled namespace
+
+Deploy the sample _bookinfo_ app from the Istio bundle.
```
Comment (rainest) [inline docs/guides/getting-started-istio.md:100]:
```suggestion
Deploy the sample _bookinfo_ app from the Istio bundle:
```

File: docs/guides/getting-started-istio.md [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+kubectl apply -n my-istio-app -f istio-1.6.7/samples/bookinfo/platform/kube/bookinfo.yaml
+```
+Wait until the application is up:
+```console
+$ kubectl wait --for=condition=Available deployment productpage -n my-istio-app --timeout=240s
+```
+### Deploy ingress
+
+Define a `KongPlugin` rate-limiting access to 100 requests per minute. Define an `Ingress` telling Kong to proxy traffic
+to a service belonging to the sample application.
```
Comment (rainest) [inline docs/guides/getting-started-istio.md:118]:
```suggestion
to a service belonging to the sample application:
```

File: docs/guides/getting-started-istio.md [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+...
+```
+
+Navigate your web browser to `http://localhost:8080/` You should be able to see a bookstore web application. Click
+through any available links several times. As you hit 30 requests per minute (for example, by holding down the "Refresh"
+key combination, e.g. `<Ctrl-R>` or `<Command-R>`), you should obtain a `Kong Error - API rate limit exceeded` response.
+
+### See the connection graph in Kiali
+
+Connect to Kiali (the Istio) dashboard:
```
Comment (rainest) [inline docs/guides/getting-started-istio.md:174]:
```suggestion
Connect to Kiali (the Istio dashboard):
```

**Comment by yangvipguang** [comment]
@hbagdi @mflendrich  hi, I haved read the doc methioned above。But It's just a simple sample。If I want to use vs or Canary deployment based on  header . How am i going to do with istio and kong ingress

**Pull request merged**